src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,70 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE;
+    private static final int DEFAULT_MULTIPLIER = 2;
+
+    private T[] elementData;
+    private int size;
+    private int currentMaxSize;
+
+    public MateStack() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        currentMaxSize = DEFAULT_CAPACITY;
+    }
+
     public void push(T value) {
+        if (value == null) {
+            return;
+        }
+        if (size == currentMaxSize) {
+            resize();
+        }
+
+        elementData[size] = value;
+        size++;
+    }
+
+    private void resize() {
+        if (size == elementData.length) {
+            int newCapacity = currentMaxSize * DEFAULT_MULTIPLIER;
+            if (newCapacity - MAX_ARRAY_SIZE > 0) {
```

Comment:
The expression 'newCapacity - MAX_ARRAY_SIZE > 0' is always false, because the maximum possible newCapacity can be MAX_ARRAY_SIZE. You should remove this redundant check.